security/signup-controller.php

```diff
@@ -0,0 +1,38 @@
+<?php
+include "db-connection.php";
+
+$conn = openCon();
+
+$firstname = isset($_POST['firstname']) ? $_POST['firstname'] : false;
+$password = isset($_POST['password']) ? $_POST['password'] : false;
+$role = isset($_POST['role']) ? $_POST['role'] : false;
```

Comment:
You repeat the same code three times - that means it must be extract in function. This will prevent future mistakes and the code will be easier to change.